src/Sparrow/Json/BlittableWriter.cs

```diff
@@ -459,10 +461,193 @@ public unsafe int WriteVariableSizeIntInReverse(int value)
             return count;
         }
 
-        public unsafe int WriteValue(string str, out BlittableJsonToken token, UsageMode mode = UsageMode.None)
+#if NET7_0_OR_GREATER
+
+        public static byte[] ByteActionTable =>
+        [
+            // 0-31: Control characters (action = 2)
+            2, 2, 2, 2, 2, 2, 2, 2, // 0-7
+            1, 1, 1, 2, 1, 1, 2, 2, // 8-15 (\b, \t, \n, _, \f, \r)
+            2, 2, 2, 2, 2, 2, 2, 2, // 16-23
+            2, 2, 2, 2, 2, 2, 2, 2, // 24-31
+
+            // 32-63: Mostly normal characters (action = 0), except " at 34
+            0, 0, 1, 0, 0, 0, 0, 0, // 32-39 (space, !, ", #, $, %, &, ')
+            0, 0, 0, 0, 0, 0, 0, 0, // 40-47 ((, ), *, +, ,, -, ., /)
+            0, 0, 0, 0, 0, 0, 0, 0, // 48-55 (0-7)
+            0, 0, 0, 0, 0, 0, 0, 0, // 56-63 (8-9, :, ;, <, =, >, ?)
+
+            // 64-95: Normal characters (action = 0), except \ at 92
+            0, 0, 0, 0, 0, 0, 0, 0, // 64-71 (@, A-G)
+            0, 0, 0, 0, 0, 0, 0, 0, // 72-79 (H-O)
+            0, 0, 0, 0, 0, 0, 0, 0, // 80-87 (P-W)
+            0, 0, 0, 0, 1, 0, 0, 0, // 88-95 (X, Y, Z, [, \, ], ^, _)
+
+            // 96-127: Normal characters (action = 0)
+            0, 0, 0, 0, 0, 0, 0, 0, // 96-103 (`, a-h)
+            0, 0, 0, 0, 0, 0, 0, 0, // 104-111 (i-p)
+            0, 0, 0, 0, 0, 0, 0, 0, // 112-119 (q-x)
+            0, 0, 0, 0, 0, 0, 0, 0, // 120-127 (y-z, {, |, }, ~, DEL)
+
+            // 128-255: UTF-8 continuation bytes or high ASCII (action = 0)
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+        ];
+
+        [SkipLocalsInit]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe int WriteValueFromStack(ReadOnlySpan<byte> str, out BlittableJsonToken token)
+        {
```

Comment:
Can we add a `Debug.Assert()` here to ensure that the size isn't too big?

Reply:
I would say we should do it like this:
```csharp
Span<int> escapePositions = str.Length < 512 ? stackalloc int[size] : new int[size];
```

I assume that scenarios where the size exceeds 512 are pretty rare, so we will have a safe path to execute it anyway instead of triggering a SOE.